app/controllers/dashboard_controller.rb

```diff
@@ -569,6 +571,10 @@ def authenticate
     end
   end
 
+  def generate_ui_api_token(userid)
+    (@ui_api_controller ||= ApiController.new).generate_user_token(userid, "ui")
```

Comment:
What do you think about putting token generation into a service object, since it's used in two different controllers?